libs/wire-api/test/unit/Test/Wire/API/Conversation.hs

```diff
@@ -0,0 +1,50 @@
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# OPTIONS_GHC -Wno-orphans #-}
+
+-- This file is part of the Wire Server implementation.
+--
+-- Copyright (C) 2020 Wire Swiss GmbH <opensource@wire.com>
+--
+-- This program is free software: you can redistribute it and/or modify it under
+-- the terms of the GNU Affero General Public License as published by the Free
+-- Software Foundation, either version 3 of the License, or (at your option) any
+-- later version.
+--
+-- This program is distributed in the hope that it will be useful, but WITHOUT
+-- ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+-- FOR A PARTICULAR PURPOSE. See the GNU Affero General Public License for more
+-- details.
+--
+-- You should have received a copy of the GNU Affero General Public License along
+-- with this program. If not, see <https://www.gnu.org/licenses/>.
+
+module Test.Wire.API.Conversation where
+
+import qualified Data.Set as Set
+import Imports
+import Test.Tasty
+import Test.Tasty.QuickCheck (testProperty, (===))
+import Wire.API.Conversation
+
+tests :: TestTree
+tests =
+  testGroup
+    "Conversation"
+    [ fromLegacyToV2ToLegacy,
+      fromV2ToLegacyToV2
```

Comment:
```suggestion
    [ acessRoleFromLegacyToV2ToLegacy,
      acessRoleFromV2ToLegacyToV2
```

Reply:
(Also, it's probably much faster if we don't run QC here but enumerate all test cases exhaustively.  But I think we should probably resist, and do that more exhaustively once we don't have anything better to do.)

Reply:
I like the renaming.
Regarding QC, all wire-api unit tests together take 1.5s to run. I don't see a problem.
Checking properties has the benefit that if we add additional access roles later, they are automatically checked.
We can't ensure exhaustiveness on sets anyway, AFAIK.